### ISSUE TITLE
[ASK] Getting recommendation with xDeepFM

### ISSUE THREAD

**Issue by AntoineBachelet** [recommenders-team/recommenders#1728]
### Description
I am actually using xDeepFM to find recommendation with implicit feedbacks and I did not find examples on how to find top-k recommendation with the model. I tried to use the function model.predict() with a file containing all the possible interactions between my users and items to find a score for each pair and sorting these scores to find top-k recommendation. This method gives me the same recommendation for every user so I would want to know if my method is wrong and what would you suggest to find top-k recommendation ?

**Comment by yueguoguo** [comment]
@Leavingseason may have some ideas about this.

**Comment by mtfs20** [comment]
Any updates on this?

**Comment by mtfs20** [comment]
> ### Description
> I am actually using xDeepFM to find recommendation with implicit feedbacks and I did not find examples on how to find top-k recommendation with the model. I tried to use the function model.predict() with a file containing all the possible interactions between my users and items to find a score for each pair and sorting these scores to find top-k recommendation. This method gives me the same recommendation for every user so I would want to know if my method is wrong and what would you suggest to find top-k recommendation ?

Did you find a way to do top-k recommendation with it?

**Comment by AntoineBachelet** [comment]
I worked on it during my internship, so I am not working on it anymore. But as far as I remember I used this technique (predict on every pair of user/item and sort to have top-k score) and acted on the different parameters of the model to have better results on my recommendations

**Comment by AntoineBachelet** [comment]
No, it seems that is was related with the parameter of the model trained (my model was underfitting) and the quality of the data I had (which was difficult to improve)

**Comment by mtfs20** [comment]
ok then i have to check my data. thanks for your help
